chore: update dashboard documentation including offline, dashboard layout and new sharing dialog (#835)
dhis2/dhis2-docs: src/user/managing-dashboards.md

#### Dashboard layout and placement of new items
When adding items to the dashboard you can choose an overall layout by clicking on **Change layout** button.
* With *Freeflow* layout, the added items can be moved using the mouse by down-clicking on the item and dragging it to the desired position. Items can also be resized with the mouse by down-clicking on the drag handle in the lower right corner of the item and dragging to the desired size.
* With *Fixed columns* layout, you can choose the number of columns to have on the dashboard, and the dashboard will automatically be layed out for you. Items cannot be moved or resized in *Fixed columns* layout. You can change this layout setting at any time.

![](resources/images/dashboard/dashboard-layout-modal.png)

You can also configure whether newly added items are automatically placed at the start or end of the dashboard by choosing the desired option. This option can be changed at any time.

![](resources/images/dashboard/dashboard-place-items.png)

#### Spacer items

> When adding items to the dashboard, they will "rise" upwards until they bump into another item. In order to force empty vertical space between items (like an empty row), you can add spacer items to the dashboard. They are only visible in edit/create mode. In view mode, they are not displayed, but take up the defined space.

Reviewer: Mention that this only applies to freeflow?
```suggestion
When adding items to the dashboard using the *Freeflow* layout, they will "rise" upwards until they bump into another item. In order to force empty vertical space between items (like an empty row), you can add spacer items to the dashboard. They are only visible in edit/create mode. In view mode, they are not displayed, but take up the defined space.
```

Reviewer: Also applies to `Removing items` below

Author: done

### Removing items

Remove items by clicking on the red trash can at the upper right of the item. Be aware that when you remove an item, the items that are positioned below the removed item will "rise" upwards until they bump into an item above.

### Print preview

> Click on the **Print preview** button to view how the dashboard would look in dashboard layout print.

Reviewer: ```suggestion
Click on the **Print preview** button to view what the dashboard will look like when being printed.
```

Reviewer: Or do you mean specifically using the "dashboard layout" mode? then maybe 
```suggestion
Click on the **Print preview** button to view what the dashboard will look like when being printed (using the *Dashboard layout* option, described further in the *Print* section below)
```

Author: done

![](resources/images/dashboard/dashboard-edit-print-preview.png)

Click on **Exit print preview** to return to editing the dashboard.

Note that some items may be moved to the next page to avoid being split between two pages. Items may also be shortened to fit on one page. Items that are shortened show an info icon in the upper right corner of the item in preview. The info icon is not visible in the actual print.

### Restricting dashboard filters { #restricting-dashboard-filters } 

By default, users will be able to filter dashboard items by any dimension defined in the DHIS2 instance. Dashboard filter settings can be edited for a given dashboard by clicking on **Filter settings**.

> ![](resources/images/dashboard/dashboard-filter-settings-button.png)

Reviewer: This screenshot (without the filter being highlighted) showcasing the buttons should ideally be placed above the Print preview section, as it applies to all of these buttons (print, filter, translate).

Author: done (see other comment related to the awkward ordering of sections)

> To restrict available filters, you can click **Only allow filtering by selected dimensions** and select the filters you wish to allow on the dashboard. Period and Organisation Unit are selected by default but can be removed if desired. When the dashboard is in view mode, users will only be able to choose from among the filters selected.

Reviewer: ```suggestion
To restrict available filters, you can click **Only allow filtering by selected dimensions** and select the filters you wish to allow on the dashboard. Period and Organisation Unit are selected by default but can be removed if desired. When the dashboard is viewed, users will only be able to choose from among the filters listed under *Selected Filters*.
```

Author: done

![](resources/images/dashboard/dashboard-filter-settings.png)

In order to save updates to filter settings, you need to first click **Confirm** to close the Filter settings dialog and then click **Save changes** to save the dashboard changes.

### Saving the dashboard

> When creating or editing a dashboard, changes are only saved when you click **Save changes** button in the dashboard edit bar at the top of the page. If you don't want to save your changes, click the **Exit without saving** button to the upper right. You will then be returned to view mode with the dashboard you were previously viewing.

Reviewer: ```suggestion
When creating or editing a dashboard, changes are only saved when you click the **Save changes** button in the dashboard edit bar at the top of the page. If you don't want to save your changes, click the **Exit without saving** button to the upper right. You will then be returned to view mode with the dashboard you were previously viewing.
```

Author: done

> ## Editing an existing dashboard

Reviewer: The order of the sections here is a bit awkward. Save (above) is in edit mode, then this section (which is really the view mode), then its followed by Translate and Delete (below) which are edit mode stuff again.
Shouldn't this section be moved down to below Delete as part of the `Viewing a dashboard` section?

Author: Yeah this has always bothered me. I've now combined Create and Edit into a single section, and just noted that Translate and Delete are only shown if editing an existing dashboard. Hopefully this makes more sense. And then before discussing the available actions, a screenshot of all the action buttons (surrounded by a red box) is shown.

> If you have access rights to edit the currently active dashboard, there will be an **Edit** button to the right of the dashboard title in view mode. Click on this button to enter edit mode.

Reviewer: Highlight `edit mode` with bold or italic? Applies to other places in the docs too.

![](resources/images/dashboard/dashboard-title-bar.png)

Refer to the above section about creating dashboards for information on making changes to the dashboard.

### Translating dashboard title and description

You can add translations for dashboard title and description while in edit mode by clicking the **Translate** button in the actions bar. The dialog provides a list of languages to translate to, and shows the original dashboard title underneath the name input field. First choose the language you want to translate for, then fill in the dashboard name and description translation.

> ![](resources/images/dashboard/dashboard-translation-dialog.png)

Reviewer: Isn't it weird how the inputs have both a label and a placeholder that says the same thing? "Name... Name". "Description... Description". Maybe this dialog is getting replaced soon anyway? @edoardo ?
![image](https://user-images.githubusercontent.com/12590483/134302327-bfe2183d-92de-4e0c-9e5d-39da58dec2bf.png)

Author: I'm sure we'll be working on a shiny new translation dialog in the near future 🤞

If you have access to delete the dashboard, then there will be a **Delete** button in the actions bar when in edit mode. A confirmation dialog will first be displayed to confirm that you want to delete the dashboard.

## Viewing a dashboard

From view mode, you can toggle showing the description, star a dashboard, apply filters, print the dashboard, make the dashboard available offline, and share the dashboard with other users and user groups.

![](resources/images/dashboard/dashboard-more-menu.png)

### Show description { #dashboard-show-description }

To toggle the description, click on the **...More** button and choose **Show description** (or **Hide description**). This setting will be remembered for all dashboards that you open. This setting applies to you, not other users.

### Star dashboards { #dashboard-star-dashboard }

Your starred dashboards are listed first in the list of dashboards. To star a dashboard, click on the star button to the right of the title. You can also toggle the star from the **...More** button. When the star is “filled”, that means the dashboard is starred. Starring a dashboard only applies to you, not other users.

### Filtering a dashboard

> Multiple filters can be applied to a dashboard for changing the data displayed in the various dashboard items. The filters are applied to each dashboard item in the same way: each added filter overrides the original value for that dimension in the original chart, table or map (visualization). It is possible to filter on Organisation Units, Periods and other dynamic dimensions depending on the DHIS2 instance.

Reviewer: why the `(visualization)` after `map`?

Reviewer: `Organisation Units, Periods and other dynamic dimensions` not sure if my English is correct here, but I interpret this as that the "other" part imples that "Org units, Periods" are "dynamic" dimensions too, which they're not.
Correct would be "A type, B type and other, C type dimensions"?

Author: changed to: `Filters can be applied to a dashboard for changing the data displayed in dashboard items that contain visualizations. The filters are applied to each dashboard item in the same way: each added filter overrides the original value for that dimension in the original chart, table or map. It is possible to filter on Organisation Units and Periods, as well as dynamic dimensions, depending on the DHIS2 instance. You can apply multiple filters to the dashboard.`

> To add a filter, click on the **Add Filter** button and choose the dimension:

Reviewer: ```suggestion
To add a filter, click on the **Add Filter** button and choose a dimension:
```

Author: done

Click on **Confirm** in the dialog to apply the filter to the
current dashboard.

> Filters are not stored, so when switching to a different dashboard they are lost. Black filter badges appear above the dashboard items to indicate that what is shown in the dashboard items is not the original visualization, but a filtered one where the filters override the stored dimensions' values.

Reviewer: ```suggestion
Filters are not stored, so when switching to a different dashboard they are reset. Black filter badges appear above the dashboard items to indicate that what is shown in the dashboard items is not the original visualization, but a filtered one where the filters override the stored dimensions' values.
```

Reviewer: Also, the sentence starting with "Black filter" is missing something. Try reading it from start to finish.

Author: Changed to `Black filter badges appear above the dashboard items to indicate which filters have been applied to the dashboard's visualizations.`

Reviewer: Sounds good. Also don't miss my first comment above, `lost` -> `reset`.

![Current filters displayed as badges above the dashboard](resources/images/dashboard/dashboard-filter-badges.png)

> Filter badges can be clicked for opening the filter selection dialogs thus allowing for filter editing. A filter can be removed by clicking on the **Remove** button in the badge. Whenever a filter is added, edited or removed, the dashboard items reload to show the updated data. The filter badges are always visible at the top of the page when scrolling the dashboard content.

Reviewer: ```suggestion
By clicking a filter badge, the filter selection dialog will open, allowing for filter editing. A filter can also be completely removed by clicking on the **Remove** button on the badge. Whenever a filter is added, edited or removed, the dashboard items reload to show the updated data. The filter badges are always visible at the top of the page when scrolling the dashboard content.
```

Author: done

> By default, users are able to filter dashboard items by any dimension defined in your system. To limit available filters, see [Restricting dashboard filters](#restricting-dashboard-filters).

Reviewer: Suggest keeping the same terminology as used in the restrict section above
```suggestion
By default, users are able to filter dashboard items by any dimension defined in the DHIS2 instance. To limit available filters, see [Restricting dashboard filters](#restricting-dashboard-filters).
```

Author: done

### Making dashboards available offline

> To make a dashboard available offline, choose the **Make dashboard available offline** option in the **...More** menu. This will cause a reload of the dashboard where requests to the server are recorded and saved in browser storage. Note that offline dashboards are only available on the computer and browser where you set it to offline. If you currently have a filter applied when requesting the dashboard be made available offline, the filter will have to be removed first.

Reviewer: For the last part, mention that this happens through a prompt?

Reviewer: i.e. 
![image](https://user-images.githubusercontent.com/12590483/134305845-0a79dec4-cd22-4c7e-a6a8-4a593113d6cf.png)

Author: done

Dashboards that have been saved for offline have an indicator on the dashboard chip in the dashboards bar, as well as a tag showing the time it was saved.

![](resources/images/dashboard/dashboard-offline-dashboard.png)

If the dashboard has been changed since you made it available offline, either by you or someone else, you'll need to choose **Sync offline data now** from the **...More** menu to save the latest version of the dashboard.

![](resources/images/dashboard/dashboard-sync-offline.png)

You can remove a dashboard from offline storaged by choosing **Remove from offline storage** in the **...More** menu.

In addition, if you manually log out of the DHIS2 instance, all cached dashboards will be cleared and they will no longer be available offline.

> ![](resources/images/dashboard/dashboard-remove-offline.png)

Reviewer: This screenshot should be moved up slightly to be next to the "You can remove a dashboard from offline...." section

Author: done

#### Other notes about Dashboards app when you are offline:
If you are offline, any buttons or actions that require a connection to complete successfully are disabled. They will be re-enabled once connectivity is restored.

### Printing a dashboard

> From the **...More** menu you can print the current dashboard. There are two styles of dashboard print: dashboard layout and one item per page. For both styles, a title page is added that shows the dashboard title, description (if the Show description setting is on), and any applied dashboard filters.

Reviewer: Highlight key concepts with italic or bold?
```suggestion
From the **...More** menu you can print the current dashboard. There are two styles of dashboard print: *Dashboard layout* and *One item per page*. For both styles, a title page is added that shows the dashboard title, description (if the *Show description* setting is enabled), and any applied dashboard filters.
```

Author: done

#### Print dashboard layout

> Dashboard layout print will approximate the dashboard layout as it is shown in the browser. Note that some adjustments may need to be made to the layout in order to prevent items being split onto multiple pages: items that would be cut off by a page break are moved to the next page, and items that are taller than one page are shortened to fit on one page.

Reviewer: ```suggestion
*Dashboard layout* print will approximate the dashboard layout as it is shown in the browser. Note that some adjustments may need to be made to the layout in order to prevent items being split onto multiple pages: items that would be cut off by a page break are moved to the next page, and items that are taller than one page are shortened to fit on one page.
```

Author: done

Click on the **Print** button in the upper right to trigger the browser print functionality.

![](resources/images/dashboard/dashboard-print-layout.png)

#### Print one item per page

This style of print will print each dashboard item on a separate page, maximizing the use of the paper size.

Click on the **Print** button in the upper right to trigger the browser print functionality.

![](resources/images/dashboard/dashboard-print-oipp.png)

## Dashboard items with charts, pivot tables or maps

Dashboard items with charts, pivot table or maps may have an item menu button in the upper right corner of the item with additional viewing options, depending on the system settings that have been configured for the DHIS2 instance. If all the relevant system settings have been disabled by the DHIS2 instance, then there will not be an item menu button. Here are the possible item menu options: 

### Switching between visualizations

Dashboard items showing charts, pivot tables and maps can be toggled

> between these visualizations. Click on the item menu button and choose the desired view (e.g., **View as Table**, **View as Map**, **View as Chart**):

Reviewer: ```suggestion
to be viewed as each of these visualization types. Click on the item menu button and choose the desired view (e.g., **View as Table**, **View as Map**, **View as Chart**):
```

Author: done

The item will be expanded vertically underneath to show the description, interpretations and replies. You can like an interpretation, reply to an interpretation, and add your own interpretation. You can edit, share or delete your own interpretations and replies, and if you have moderator access, you can delete others’ interpretations.

It is possible to format the description field, and interpretations with **bold**, _italic_ by using the Markdown style markers \* and \_ for **bold** and _italic_ respectively. The text field for writing new interpretations has a toolbar for adding rich text. Keyboard shortcuts are also available: Ctrl/Cmd + B and Ctrl/Cmd + I. A limited set of smilies is supported and can be used by typing one of the following character combinations: :) :-) :( :-( :+1 :-1. URLs are automatically detected and converted into a clickable link.

Interpretations are sorted in descending order by date, with the most recent shown on top. Interpretation replies are sorted in ascending order by date, with the oldest shown on top.

![](resources/images/dashboard/dashboard-interpretations.png)

## Sharing a dashboard { #dashboard_sharing } 

> In order to share a dashboard with users and user groups, click on the **Share** button to the right of the dashboard title to display the *Sharing & Access* dialog.

Reviewer: why does the modal say `Sharing & Access` with a capital A? 🤔

Author: Will be updated in the ui component

![](resources/images/dashboard/dashboard-sharing-dialog.png)

There are 3 levels of sharing permissions available for a dashboard:

- No access

  The user or user group cannot view or edit the dashboard. If they try to access the dashboard with the url, the Dashboards app will display the message "Requested dashboard not found".

- View only

  The user or user group can view the dashboard but cannot edit it.

- View and edit

>   The user or user group can view and edit the dashboard. Editing allows for altering the layout, resizing and removing items, renaming and deleting the dashboard, etc.

Reviewer: ```suggestion
  The user or user group can view and edit the dashboard. This allows for all types of changes, such as altering the layout, resizing and removing items, renaming and deleting the dashboard, etc.
```

Author: done

All dashboards have two sharing groups set by default.

- External users

  The *External users* group defines access to the dashboard for people not logged in to the DHIS2 instance. By default the setting is **No access** and can be changed to **View only**. Giving access to external users is useful for when you are creating an external web portal and would like to present information from DHIS2.

- All users

  The *All users* group defines access to the dashboard for all logged-in users in the DHIS2 instance. By default the setting is **No access** and can be changed to **View only** or **View and edit**.

### Add users and user groups

To share a dashboard with specific users and user groups, type the name in the input field, choose the desired access level and click on **Give access**.

> ![](resources/images/dashboard/dashboard-sharing-add-user.png)

Reviewer: oh how the OCD alert goes off when the first screenshot has a dashboard named "test" and the second one is named "Test" 
![image](https://user-images.githubusercontent.com/12590483/134308949-fd40ff51-748e-4180-8ea8-a7f037bd1aef.png)
![image](https://user-images.githubusercontent.com/12590483/134309008-634016a7-67eb-4bee-8cde-568b7539dadf.png)

You can provide users with the url of the dashboard, allowing them to navigate directly to the dashboard. To get the dashboard url, just open the dashboard in view mode, and copy the browser url. For example, the url to the Antenatal Care dashboard in play.dhis2.org/demo is:

> https://play.dhis2.org/demo/dhis-web-dashboard/\#/nghVC4wtyzi

Reviewer: there's a backslash just before the hashtag, is that for escaping something in our parsers? either way, its not the correct url I suppose but it still works when clicking it (takes you to https://play.dhis2.org/2.36.3/dhis-web-dashboard/#/nghVC4wtyzi though....)

Author: Changing the url to specifically say "dev" and remove the backslash. I'll need to go into the older versions and switch the url to 2.36, 2.35, and 2.34.

### Cascade sharing of visualizations on the dashboard

> ![](resources/images/dashboard/dashboard-sharing-cascade-sharing.png)

Reviewer: Will there be more coming here? No text, just a screenshot 😿

Author: Yeah I need to fill this in. Need to talk to Jan first.
